presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewriter.java

```diff
@@ -0,0 +1,227 @@
+
+        if (!tableLayoutHandle.isPresent()) {
+            return createEmptyStatsTable(table.getName());
+        }
+
+        TableStatistics tableStatistics = metadata.getTableStatistics(session, tableHandle, tableLayoutHandle.get());
+
+        List<String> statisticsNames = findUniqueStatisticsNames(tableStatistics);
+
+        List<String> resultColumnNames = buildColumnsNames(statisticsNames);
+        SelectItem[] selectItems = buildSelectItems(resultColumnNames);
+
+        Map<ColumnHandle, String> columnNames = getStatisticsColumnNames(tableStatistics, node, table.getName());
+        List<Expression> resultRows = buildStatisticsRows(tableStatistics, columnNames, statisticsNames);
+
+        return simpleQuery(selectList(selectItems),
+                aliased(new Values(resultRows),
+                        "table_stats_for_" + table.getName(),
+                        resultColumnNames));
+    }
+
+    private Optional<TableLayoutHandle> getTableLayoutHandle(ShowStats node, TableHandle tableHandle)
+    {
+        Optional<TableLayoutHandle> handle = findTableLayoutHandleWithWhere(node);
+        if (!handle.isPresent()) {
+            return findEmptyConditionLayoutHandle(tableHandle);
+        }
+        return handle;
+    }
+
+    private Optional<TableLayoutHandle> findEmptyConditionLayoutHandle(TableHandle tableHandle)
+    {
+        List<TableLayoutResult> layouts = metadata.getLayouts(session, tableHandle, Constraint.alwaysTrue(), Optional.empty());
+        if (layouts.isEmpty()) {
+            return Optional.empty();
+        }
+        return Optional.of(layouts.get(0).getLayout().getHandle());
+    }
+
+    private Optional<TableLayoutHandle> findTableLayoutHandleWithWhere(ShowStats node)
+    {
+        checkArgument(node.getQuery().getQueryBody() instanceof QuerySpecification);
+        QuerySpecification specification = (QuerySpecification) node.getQuery().getQueryBody();
+
+        if (!specification.getWhere().isPresent()) {
+            return Optional.empty();
+        }
+
+        Plan plan = queryExplainer.getLogicalPlan(session, node.getQuery(), parameters);
+        Map<Symbol, Type> types = plan.getTypes();
+
+        Optional<TableScanNode> scanNode = findTableScanNode(plan);
+        if (!scanNode.isPresent()) {
+            return Optional.empty();
+        }
+
+        Expression filterNodePredicate = findFilterExpression(plan);
+
+        List<TableLayoutResult> tableLayouts = getLayouts(scanNode.get(), filterNodePredicate, types);
+
+        if (tableLayouts.isEmpty()) {
+            return Optional.empty();
+        }
+
+        return Optional.of(tableLayouts.get(0).getLayout().getHandle());
+    }
+
+    private Node createEmptyStatsTable(QualifiedName tableName)
+    {
+        return simpleQuery(selectList(unaliasedName("column_name")), aliased(new Values(ImmutableList.of()), "table_stats_for_" + tableName, ImmutableList.of("column_name")));
```

Comment:
As we discussed. IMO this method should be called `createStatsForEmptyLayout` and return table filled with zeros.